compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2043,17 +2045,39 @@ public void visit(BLangRecordLiteral recordLiteral) {
                     continue;
                 }
 
-                for (BField bField : ((BRecordType) spreadOpExpr.type).fields.values()) {
-                    if (Symbols.isOptional(bField.symbol)) {
-                        continue;
+                if (!((BRecordType) spreadOpExpr.type).sealed) {
+                    // More than one inclusive-type-descriptors with spread-fields are not allowed.
+                    if (noOfInclusiveTypes > 0) {
+                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.MORE_THAN_ONE_INCLUSIVE_TYPES,
+                                recordLiteral.expectedType.getKind().typeName());
+                    } else {
+                        noOfInclusiveTypes = noOfInclusiveTypes + 1;
+                    }
+                }
+
+                LinkedHashMap<String, BField> fieldsInRecordLiteral = ((BRecordType) spreadOpExpr.type).fields;
+                for (Object fieldName : names) {
+                    if (fieldsInRecordLiteral.containsKey(fieldName)) {
+                        if (fieldsInRecordLiteral.get(fieldName).type.tag != TypeTags.NEVER) {
+                            // An error is logged if the same field is occurred, and it is not defined as a never-type
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL_SPREAD_OP,
+                                    recordLiteral.expectedType.getKind().typeName(), fieldName, spreadOpField);
+                        }
+                    } else {
+                        if (!((BRecordType) spreadOpExpr.type).sealed) {
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.INCLUSIVE_RECORD_LITERAL_WITH_SPREAD_OP,
+                                    recordLiteral.expectedType.getKind().typeName(), spreadOpExpr);
+                        }
                     }
+                }
 
+                for (BField bField : ((BRecordType) spreadOpExpr.type).fields.values()) {
```

Comment:
1. Do we need two for loops here?
2. I believe logic assumes we can analyze this iteratively. But I think we should think about final resultant mapping constructor expression. So we may need two passes and maybe three to give errors as well.
#### Case A: Errors and duplicate errors
```Ballerina
import ballerina/io;

type Foo record {
    int i;
};

type Bar record {|
    int j;
    int k;
|};

public function main() {
    Foo f = {i: 1, "k": 7};
    Bar b = {j: 2, "k": 4};

    map<anydata> m = {...b, ...f};    // b first
    io:println("m : ", m);
}
```
```LogTalk
        case5.bal
error: .::case5.bal:16:32: invalid usage of map literal: inclusive record 'f' with spread operator is not allowed
error: .::case5.bal:16:32: invalid usage of map literal: inclusive record 'f' with spread operator is not allowed
```

#### Case B: No errors (Same code with a different order in spread operator fields)

```Ballerina
import ballerina/io;

type Foo record {
    int i;
};

type Bar record {|
    int j;
    int k;
|};

public function main() {
    Foo f = {i: 1, "k": 7};
    Bar b = {j: 2, "k": 4};

    map<anydata> m = {...f, ...b};  // a first
    io:println("m : ", m);
}
```
```LogTalk
m : i=1 k=4 j=2
```